lib/iris/tests/test_whatsnew_contributions.py

```diff
@@ -0,0 +1,65 @@
+# (C) British Crown Copyright 2010 - 2015, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Check the the latest "whatsnew" contributions files have usable names.
+
+The files in "./docs...whatsnew/contributions_<xx.xx>/" should have filenames
+with a particular structure, which encodes a summary name.
+These names are interpreted by "./docs...whatsnew/aggregate_directory.py".
+This test just ensures that all those files have names which that process
+can accept.
+
+"""
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+# import iris tests first.
+import iris.tests as tests
+
+import imp
+import os
+import os.path
+
+
+class TestWhatsnewContribs(tests.IrisTest):
+    def test_check_contributions(self):
+        # Get dirpath of overall iris installation.
+        # Note: assume iris at "<install>/lib/iris", docs at "<install>/docs".
+        import iris
+        _iris_module_dirpath = os.path.dirname(iris.__file__)
+        _iris_dirs = _iris_module_dirpath.split(os.sep)
+        _install_dirpath = os.sep.join(_iris_dirs[:-2])
+
+        # Construct path to docs 'whatsnew' directory.
+        _whatsnew_dirpath = os.sep.join(
+            (_install_dirpath, 'docs', 'iris', 'src', 'whatsnew'))
+
+        # Tweak the imports path and import the whatsnew-aggregation code.
+        import sys
+        old_paths = sys.path
+        try:
+            sys.path.append(_whatsnew_dirpath)
+            import aggregate_directory
+        finally:
+            sys.path = old_paths
+
+        # Run the check routine.
+        aggregate_directory.check_all_contributions_valid()
```

Comment:
NB.

1. A check/validate mode would be handy for contributors anyway.
2. It would avoid the need to expose code in aggregate_directory which is only used in the tests.